dart-impl/mpi/src/dart_communication.c

```diff
@@ -193,19 +215,45 @@ dart_ret_t dart_get(
                    nelem, (unsigned long)win, team_unit_id.id, offset, dest);
   }
 
-
-  DART_LOG_TRACE("dart_get:  MPI_Get");
-  if (MPI_Get(dest,
-              nelem,
-              mpi_dtype,
-              team_unit_id.id,
-              offset,
-              nelem,
-              mpi_dtype,
-              win)
-      != MPI_SUCCESS) {
-    DART_LOG_ERROR("dart_get ! MPI_Rget failed");
-    return DART_ERR_INVAL;
+  /*
+   * MPI uses offset type int, do not copy more than INT_MAX elements:
+   */
+  // chunk up the get
+  const size_t nchunks   = nelem / INT_MAX;
```

Comment:
It has actually arrived in modern age since MPI-3, see the official [Ticket](https://github.com/mpi-forum/mpi-forum-historic/issues/117). The problem are the legacy applications needing backward compatibility.